taskvine/src/manager/vine_manager.c

```diff
@@ -2723,6 +2725,10 @@ static void commit_task_to_worker(struct vine_manager *q, struct vine_worker_inf
 	if (result != VINE_SUCCESS) {
 		debug(D_VINE, "Failed to send task %d to worker %s (%s).", t->task_id, w->hostname, w->addrport);
 		handle_failure(q, w, t, result);
```

Comment:
Double check this: under certain circumstances, `handle_failure` can call `reap_task_from_worker` and so you might be a `function_slots_inuse` decrement without a corresponding increment.

Reply:
You're right. The solution is to increment the function slots no matter what, then `handle_failure` can decrement it if failure happens.